### PR TITLE
feat: add getRequestedCredentialsForProofRequest 

### PR DESCRIPTION
**Pull request by NB-MikeRichardson**
The method getRequestedCredentialsForProofRequest was present in the API before the PPV2 code was merged but somehow got lost.

It has been restored in this PR

### CODE REVIEW

File: packages/core/src/modules/proofs/ProofsApi.ts [hunk 1]
```diff
@@ -70,6 +70,11 @@ export interface ProofsApi<PFs extends ProofFormat[], PSs extends ProofService<P
     options: AutoSelectCredentialsForProofRequestOptions
   ): Promise<FormatRequestedCredentialReturn<PFs>>
 
+  // Get Requested Credentials
+  getRequestedCredentialsForProofRequest(
+    options: AutoSelectCredentialsForProofRequestOptions
```
Comment (TimoGlastra) [inline packages/core/src/modules/proofs/ProofsApi.ts:75]: Can we define a custom type for this? This type is meant for the auto select method

File: yarn.lock [hunk 2]
```diff
@@ -34,38 +34,38 @@
   dependencies:
     "@babel/highlight" "^7.18.6"
 
-"@babel/compat-data@^7.17.7", "@babel/compat-data@^7.18.8":
```
Comment (TimoGlastra) [inline yarn.lock:37]: Changes to this file seems unrelated to this pr?

**Review by TimoGlastra — COMMENTED** [review]
Could you add a test for this flow?

**Comment by NB-MikeRichardson** [comment]
Comments resolved, a new e2e test added, yarn.lock corrected.

**Review by TimoGlastra — APPROVED** [review]
